web/src/stream_data.ts

```diff
@@ -271,9 +282,9 @@ export function slug_to_stream_id(slug: string): number | undefined {
     */
 
     // "New" (2018) format: ${stream_id}-${stream_name} .
-    const match = /^(\d+)(-.*)?$/.exec(slug);
+    const match = /^(\d+)(?:-.*)?$/.exec(slug);
     const newFormatStreamId = match ? Number.parseInt(match[1]!, 10) : null;
-    if (newFormatStreamId && stream_info.get(newFormatStreamId)) {
+    if (newFormatStreamId !== null && stream_info.get(newFormatStreamId)) {
```

Comment:
Is it intentional that this reworks the new `slug_to_stream_id` function? Might be helpful to have that in its own commit.

Reply:
Yeah this came from previous review comments:
* https://github.com/zulip/zulip/pull/31299#discussion_r1722270902
* https://github.com/zulip/zulip/pull/31299#discussion_r1722276031

Separating it into its own commit sounds good.